lib/auth/password.go

```diff
@@ -37,19 +39,45 @@ type ChangePasswordWithTokenRequest struct {
 	U2FRegisterResponse *u2f.RegisterChallengeResponse `json:"u2f_register_response,omitempty"`
 }
 
-// ChangePasswordWithToken changes password with token
-func (s *Server) ChangePasswordWithToken(ctx context.Context, req ChangePasswordWithTokenRequest) (types.WebSession, error) {
+// ChangePasswordWithToken changes password with a password reset token.
+func (s *Server) ChangePasswordWithToken(ctx context.Context, req *proto.ChangePasswordWithTokenRequest) (*proto.ChangePasswordWithTokenResponse, error) {
 	user, err := s.changePasswordWithToken(ctx, req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	sess, err := s.createUserWebSession(ctx, user)
+	var recoveryCodes []string
+	recoveryAllowed := false
+
+	// Only user's with email as their username and running cloud can receive recovery codes.
+	if _, err := mail.ParseAddress(user.GetName()); err == nil {
+		if err := s.isAccountRecoveryAllowed(ctx); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		recoveryAllowed = true
+	}
```

Comment:
this part is not clear, lets refactor it to something that is easier to follow:

```golang
if shouldCreateRecoveryCodes {
  //
}
```

Reply:
renamed